src/search/screens/search.screen.js

```diff
@@ -155,6 +154,19 @@ class Search extends Component {
     this.renderItem = this.renderItem.bind(this);
   }
 
+  componentDidMount() {
+    this.setQueryAsLocation();
+  }
+
+  setQueryAsLocation = () => {
+    const { params } = this.props.navigation.state;
+
+    if (params) {
```

Comment:
We are already using navigation state in this screen to store/retrieve the searched keyword (see line 38 for example).
Here, it seems that as soon as there's a parameter, you assume that it's related to location, which seems wrong.